azure-functions/acbs-function/mappings/facility/facility-covenant-amend.test.ts

```diff
@@ -0,0 +1,57 @@
+import { facilityCovenantAmend } from './facility-covenant-amend';
+
+describe('facilityCovenantAmend', () => {
+  const aValidAmount = 123.456;
+  const aValidAmountExpected = 123.46;
+  const aValidGuaranteeExpiryDate = '2022-01-01';
+  const aValidGuaranteeExpiryDateExpected = '2022-01-01';
+  const aValidAmountRequest = { amount: aValidAmount };
+  const aValidAmountExpectedResult = { targetAmount: aValidAmountExpected };
+  const aValidGuaranteeExpiryDateRequest = { facilityGuaranteeDates: { guaranteeExpiryDate: aValidGuaranteeExpiryDate } };
+  const aValidGuaranteeExpiryDateExpectedResult = { expirationDate: aValidGuaranteeExpiryDateExpected };
+  const aValidAmountAndGuaranteeExpiryDateRequest = { ...aValidAmountRequest, ...aValidGuaranteeExpiryDateRequest };
+  const aValidAmountAndGuaranteeExpiryDateExpectedResult = { ...aValidAmountExpectedResult, ...aValidGuaranteeExpiryDateExpectedResult };
```

Comment:
I'm not sure if this is an established pattern/approach (in which case we should stick to it), otherwise maybe could this more object structured, then the test specs below have (IMO) shorter and easier to read variables e.g

```ts
const valid = {
  amount: 123.456,
  guaranteeExpiryDate: '2022-01-01',
};

const expected = {
  amount: 123.46,
  guaranteeExpiryDate: '2022-01-01',
};

// ...

describe('when there is a guarantee expiry date', () => {
  it('returns the guarantee expiry date', () => {
    const result = facilityCovenantAmend(valid.guaranteeExpiryDateRequest);
    expect(result).toEqual(expected.guaranteeExpiryDate);
  });
});
```

What do you think?

Reply:
I've removed unneeded variables and updated the formatting -- hopefully it reads easier! Let me know

Reply:
Just to jump in on this thread, the `a<Thing>` pattern is something we have started doing in E2ETF. The reason behind it is that, in the code extract above, `const valid = { ... }` doesn't actually prevent you from overwriting something inside the object. For example, the below code is perfectly fine even in typescript

```typescript
describe('when I am testing something', () => {
  const valid = {
    amount: 123.456,
    guaranteeExpiryDate: '2022-01-01',
  };
  
  it('tests that we get the valid object back', () => {
    const result = myFunction(); // returns { amount: 654.321, guaranteeExpiryDate: '2022-01-01' }
    
    valid.amount = 654.321; // modified so that the test passes
    expect(result).toEqual(valid);
  });
  // value of 'valid' has now been overwritten for all other tests
});
```

In our tests on E2ETF, we would do something closer to the following:

```typescript
describe('when I am testing something', () => {
  const aValidResult = () => ({
    amount: 123.456,
    guaranteeExpiryDate: '2022-01-01',
  });
  
  it('tests that we get the valid object back', () => {
    const result = myFunction(); // returns { amount: 654.321, guaranteeExpiryDate: '2022-01-01' }
    
    const valid = { ...aValidResult(), amount: 654.321 };
    expect(result).toEqual(valid); // passes
  });
  // when 'aValidResult()' is called in other tests, you get the result you expect
});
```

Obviously the examples above are not very well written tests, but it is quite easy to accidentally overwrite a field in an object, and mitigating the gotcha by use of a function can make it easier to catch errors in tests